app/Entirety/semantics/views.py

```diff
@@ -15,9 +15,11 @@ class SemanticsVisualizer(ProjectContextMixin, TemplateView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         # PrepData.generate_df(self)
-        context['elements'] = PrepData.generate_df(self)
+        context['elements'] = PrepData.generate_df(self)[0]
         context['types'] = PrepData.types(self)
         context['relationships'] = PrepData.relationships(self)
+        context['entity_ids'] = PrepData.generate_df(self)[1]
+        context['entity_names'] = PrepData.generate_df(self)[2]
```

Comment:
Is it necessary to invoke the `generate_df` for three times?

Reply:
No its not, good catch. Changed it